src/dawn/CodeGen/GridTools/GTCodeGen.cpp

```diff
@@ -465,6 +469,14 @@ GTCodeGen::generateStencilInstantiation(const StencilInstantiation* stencilInsta
           const auto& field = fields[accessorIdx];
           std::string paramName = stencilInstantiation->getNameFromAccessID(field.getAccessID());
 
+          // intialize map entry      
+          collectedExtents[paramName] = std::vector<int>(field.getExtents().getSize()*2);
+          // get largest and smallest extent for each field
+          for(int dim = 0; dim < field.getExtents().getSize(); ++dim) {
```

Comment:
But anyhow as said, I would put this logic in a method of Stencil.h, which I would call from here.